core/src/main/java/org/sonarsource/sonarlint/core/notifications/NotificationChecker.java

```diff
@@ -0,0 +1,132 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.notifications;
+
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.annotation.CheckForNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.api.utils.DateUtils;
+import org.sonarsource.sonarlint.core.client.api.notifications.SonarQubeNotification;
+import org.sonarsource.sonarlint.core.container.connected.SonarLintWsClient;
+import org.sonarsource.sonarlint.core.container.model.DefaultSonarQubeNotification;
+import org.sonarsource.sonarlint.core.util.StringUtils;
+import org.sonarsource.sonarlint.core.util.ws.WsResponse;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+class NotificationChecker {
+  private static final Logger LOG = LoggerFactory.getLogger(NotificationChecker.class);
+  private static final String API_PATH = "api/developers/search_events";
+  private static final DateTimeFormatter TIME_FORMATTER = DateTimeFormatter.ofPattern(DateUtils.DATETIME_FORMAT);
+
+  private final SonarLintWsClient wsClient;
+
+  NotificationChecker(SonarLintWsClient wsClient) {
+    this.wsClient = wsClient;
+  }
+
+  /**
+   * Get all notification events for a set of projects after a given timestamp. 
+   * Returns an empty list if an error occurred making the request or parsing the response.
+   */
+  @CheckForNull
+  public List<SonarQubeNotification> request(Map<String, ZonedDateTime> requestConfig) {
+    String path = getPath(requestConfig);
+    WsResponse wsResponse = wsClient.rawGet(path);
+    if (!wsResponse.isSuccessful()) {
+      LOG.debug("Failed to get notifications: {}, {}", wsResponse.code(), wsResponse.content());
+      return Collections.emptyList();
+    }
+
+    return parseResponse(wsResponse.content());
+  }
+
+  /**
+   * Checks whether a server supports notifications
+   */
+  public boolean isSupported() {
+    String path = getPath(Collections.emptyMap());
+    WsResponse wsResponse = wsClient.rawGet(path);
+    return wsResponse.isSuccessful();
+  }
+
+  private static String getPath(Map<String, ZonedDateTime> requestConfig) {
+    StringBuilder builder = new StringBuilder();
+    builder.append(API_PATH);
+
+    builder.append("?projects=");
+    builder.append(requestConfig.keySet().stream()
+      .map(StringUtils::urlEncode)
+      .collect(Collectors.joining(",")));
+
+    builder.append("&from=");
+    builder.append(requestConfig.values().stream()
+      .map(timestamp -> timestamp.format(TIME_FORMATTER))
+      .map(StringUtils::urlEncode)
+      .collect(Collectors.joining(",")));
+
+    // TODO REMOVE THIS
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135)